### PR TITLE
Fix: Closing network relay popup without pressing a button left connection in limbo.

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

When connecting to a server that requires a relay, a popup is shown to the user, if this popup is closed without pressing an action button (e.g. by clicking on the X or right-click close) this leaves the game connection in limbo, as it cannot proceed. It will eventually time out.

## Description

Treat closing the window the same as clicking on "No".

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/network/network_gui.cpp [hunk 1]
```diff
@@ -2413,6 +2413,26 @@ struct NetworkAskRelayWindow : public Window {
 		this->InitNested(0);
 	}
 
+	void Close([[maybe_unused]] int data = 0) override
```
Comment (TrueBrain) [inline src/network/network_gui.cpp]: not "maybe" unused. Very used :)

What I find difficult to see now, is whether `WID_NAR_YES_ONCE` for example is not also `0`. How do I know that this default value is not used?

Maybe we should use `-1` as default?

Comment (PeterN) [reply src/network/network_gui.cpp]: Probably passing the widget index here is not correct. Defaulting to -1 would be a design pattern change from how OnInvalidateData is used. 

Given we also need to handle CloseWindowByClass() which should not issue ConnectFailure, and is definitely not a widget, I think it makes sense to revert the last commit here and then pass handled/unhandled state instead.

Comment (TrueBrain) [reply src/network/network_gui.cpp]: Still this pesky `maybe_unused` that is annoying me; nice solution otherwise :)

Comment (PeterN) [reply src/network/network_gui.cpp]: Hmm, well, I followed the pattern used elsewhere maybe they are wrong too?

Comment (TrueBrain) [reply src/network/network_gui.cpp]: We really should only add it in places where it is unused, in my opinion. Adding it to places where it is used, is weird?

The parent can still have it, and all other definitions of `Close`, but here it is clearly used? Feels weird to keep it. Before you know it, we just add it everywhere :D

**Review by TrueBrain — APPROVED** [review]

**Pull request merged**